<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Nan Xiao &lt;https://nanx.me&gt;  Gao Wang &lt;gaow@uchicago.edu&gt;  Lei Sun &lt;sunl@uchicago.edu&gt;" />

<meta name="date" content="2018-04-23" />

<title>Gene ID Mapping for Genotype-Tissue Expression (GTEx) Data</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 800px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
font-size: 16px;
line-height: 24px;
color: #333;
}
h1.title {
text-align: center;
font-weight: 600;
}
h4.author {
text-align: center;
}
h4.date {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 0px;
padding: 15px;
width: 770px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 16px;
line-height: 1.5;
}
#TOC .toctitle {
font-weight: bold;
font-size: 16px;
margin-left: 5px;
}
#TOC ul {
padding-left: 50px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 20px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 5px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 5px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Menlo, Monaco, "Lucida Console", Consolas, "Courier New", monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 25px;
font-size: 35px;
line-height: 40px;
font-weight: 600;
}
h2 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 1px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
margin-left: 8px;
font-size: 110%;
font-weight: normal;
}
em {
font-style: normal;
}
emph {
font-style: oblique;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #123d79;
text-decoration: none;
}
a:hover {
color: #007CC3; }
a:visited {
color: #581858; }
a:visited:hover {
color: #007CC3; }

code > span.kw { color: #4271ae; } 
code > span.dt { color: #c82829; } 
code > span.dv { color: #f5871f; } 
code > span.bn { color: #718c00; } 
code > span.fl { color: #718c00; } 
code > span.ch { color: #718c00; } 
code > span.st { color: #718c00; } 
code > span.co { color: #8e908c; } 
code > span.ot { color: #4d4d4c; } 
code > span.al { color: #ff0000; } 
code > span.fu { color: #4271ae; } 
code > span.er { color: #a61717; } 

img {
display: block;
margin: 0 auto;
}
</style>

</head>

<body>




<h1 class="title toc-ignore">Gene ID Mapping for Genotype-Tissue Expression (GTEx) Data</h1>
<h4 class="author"><em>Nan Xiao &lt;<a href="https://nanx.me" class="uri">https://nanx.me</a>&gt;<br> Gao Wang &lt;<a href="mailto:gaow@uchicago.edu">gaow@uchicago.edu</a>&gt;<br> Lei Sun &lt;<a href="mailto:sunl@uchicago.edu">sunl@uchicago.edu</a>&gt;</em></h4>
<h4 class="date"><em>2018-04-23</em></h4>


<div id="TOC">
<ul>
<li><a href="#introduction"><span class="toc-section-number">1</span> Introduction</a></li>
<li><a href="#mapping-table"><span class="toc-section-number">2</span> Mapping Table</a></li>
<li><a href="#code-example"><span class="toc-section-number">3</span> Code Example</a></li>
<li><a href="#whats-next"><span class="toc-section-number">4</span> What’s Next?</a></li>
<li><a href="#acknowledgements"><span class="toc-section-number">5</span> Acknowledgements</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The Genotype-Tissue Expression (GTEx) project <span class="citation">(Lonsdale et al. 2013)</span> aims at measuring human tissue-specific gene expression levels. With the collected data, we will be able to explore the landscape of gene expression, gene regulation, and their deep connections with genetic variations.</p>
<p>Raw GTEx data contains expression measurements from various types of elements (such as genes, pseudogenes, noncoding DNA sequences) covering the whole genome. For some analysis, it might be desirable to only keep a subset of the data, for example, data from protein coding genes. In such cases, mapping the original Ensembl gene IDs to Entrez gene IDs or HGNC symbols become an essential step in the analysis pipeline.</p>
<p><code>grex</code> offers a <strong>minimal dependency</strong> solution to do such ID mappings. Currently, an Ensembl ID from GTEx can be mapped to its Entrez gene ID, HGNC gene symbol, and UniProt ID, with basic annotation information such as HGNC gene name, cytogenetic location, and gene type. We also limit our scope on the Ensembl IDs appeared in the gene read count data. Ensembl IDs from transcript data will be considered in future versions.</p>
</div>
<div id="mapping-table" class="section level1">
<h1><span class="header-section-number">2</span> Mapping Table</h1>
<p>To facilitate such ID conversion tasks, the <code>grex</code> package has a built-in mapping table derived from the well-known annotation data package <code>org.Hs.eg.db</code> <span class="citation">(Carlson 2015)</span>. The mapping data we used has integrated mapping information from Ensembl and NCBI, to maximize the possibility of finding a matched Entrez ID. The R script for creating the mapping table is located <a href="https://github.com/road2stat/grex/blob/master/data-raw/mapping-table-generator.R">here</a>.</p>
<p>Not surprisingly, when creating such a table, there were hundreds of cases where a single Ensembl ID can be mapped to multiple Entrez gene IDs. To create a one-to-one mapping, we took a simple approach: we just removed the duplicated Entrez IDs and only kept the first we encountered in the original database. Therefore, there might be cases where the mapping is not 100% accurate. If you have such doubts for particular results, please try searching the original ID on the Ensembl website and see if we got a correct mapped ID.</p>
</div>
<div id="code-example" class="section level1">
<h1><span class="header-section-number">3</span> Code Example</h1>
<p>As an example, we use the Ensembl IDs from GTEx V7 gene count data and select 100 IDs:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(<span class="st">&quot;grex&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">data</span>(<span class="st">&quot;gtexv7&quot;</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">id =<span class="st"> </span>gtexv7[<span class="dv">101</span><span class="op">:</span><span class="dv">200</span>]</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">df =<span class="st"> </span><span class="kw">grex</span>(id)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="kw">tail</span>(df)</a></code></pre></div>
<pre><code>##          ensembl_id entrez_id hgnc_symbol                                   hgnc_name
## 95  ENSG00000266075      &lt;NA&gt;        &lt;NA&gt;                                        &lt;NA&gt;
## 96  ENSG00000272153      &lt;NA&gt;        &lt;NA&gt;                                        &lt;NA&gt;
## 97  ENSG00000116198      9731      CEP104                     centrosomal protein 104
## 98  ENSG00000169598      1677        DFFB       DNA fragmentation factor subunit beta
## 99  ENSG00000198912    339448    C1orf174         chromosome 1 open reading frame 174
## 100 ENSG00000236423 100133612   LINC01134 long intergenic non-protein coding RNA 1134
##     cyto_loc uniprot_id   gene_biotype
## 95      &lt;NA&gt;       &lt;NA&gt;           &lt;NA&gt;
## 96      &lt;NA&gt;       &lt;NA&gt;           &lt;NA&gt;
## 97   1p36.32     O60308 protein_coding
## 98   1p36.32     B4DZS0 protein_coding
## 99   1p36.32     Q8IYL3 protein_coding
## 100  1p36.32       &lt;NA&gt;        lincRNA</code></pre>
<p>The elements which cannot be mapped accurately will be <code>NA</code>.</p>
<p>Genes with a mapped Entrez ID:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">filtered_genes =</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="st">  </span>df[<span class="op">!</span><span class="kw">is.na</span>(df<span class="op">$</span><span class="st">&quot;entrez_id&quot;</span>),</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">     <span class="kw">c</span>(<span class="st">&quot;ensembl_id&quot;</span>, <span class="st">&quot;entrez_id&quot;</span>, <span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;gene_biotype&quot;</span>)]</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">head</span>(filtered_genes)</a></code></pre></div>
<pre><code>##        ensembl_id entrez_id hgnc_symbol         gene_biotype
## 1 ENSG00000162576     54587       MXRA8       protein_coding
## 2 ENSG00000175756     54998    AURKAIP1       protein_coding
## 4 ENSG00000221978     81669       CCNL2       protein_coding
## 5 ENSG00000224870    148413   LOC148413 processed_transcript
## 6 ENSG00000242485     55052      MRPL20       protein_coding
## 8 ENSG00000235098    441869     ANKRD65       protein_coding</code></pre>
<p>If you want to start from the raw GENCODE gene IDs provided by GTEx (e.g. <code>ENSG00000227232.4</code>), the function <code>cleanid()</code> can help you remove the <code>.version</code> part in them, to produce Ensembl IDs.</p>
</div>
<div id="whats-next" class="section level1">
<h1><span class="header-section-number">4</span> What’s Next?</h1>
<p>Conventionally, the next step is removing (or imputing) the genes with <code>NA</code> IDs, and then select the genes to keep. Notably, as was observed in the complete gene read count data, in about 100 cases, multiple Ensembl IDs can be mapped to one single Entrez ID. Post-processing steps may also be needed for such genes.</p>
</div>
<div id="acknowledgements" class="section level1">
<h1><span class="header-section-number">5</span> Acknowledgements</h1>
<p>We thank members of the <a href="http://stephenslab.uchicago.edu">Stephens lab</a> (Kushal K Dey, Michael Turchin) for their valuable suggestions and helpful discussions on this problem.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-hsdb">
<p>Carlson, Marc. 2015. <em>org.Hs.eg.db: Genome Wide Annotation for Human</em>.</p>
</div>
<div id="ref-gtex">
<p>Lonsdale, John, Jeffrey Thomas, Mike Salvatore, Rebecca Phillips, Edmund Lo, Saboor Shad, Richard Hasz, et al. 2013. “The Genotype-Tissue Expression (GTEx) Project.” <em>Nature Genetics</em> 45 (6). Nature Publishing Group:580–85.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
